state/src/interface/task_queue.rs

```diff
@@ -167,15 +167,26 @@ impl State {
     }
 
     /// Pause a task queue placing the given task at the front of the queue
-    #[instrument(name = "propose_preempt_task_queue", skip_all, err, fields(task_id, task = %task.display_description()))]
     pub async fn pause_task_queue(
         &self,
         key: &TaskQueueKey,
         task_id: TaskIdentifier,
         task: TaskDescriptor,
     ) -> Result<ProposalWaiter, StateError> {
-        // Pick a task ID and create a task from the description
-        backfill_trace_field("task_id", task_id.to_string());
+        self.pause_multiple_task_queues(vec![*key], task_id, task).await
+    }
+
+    /// Pause multiple task queues, placing the given task at the front of each
+    /// queue
+    #[instrument(name = "propose_preempt_task_queues", skip_all, err, fields(queue_keys, task_id = %task_id, task = %task.display_description()))]
+    pub async fn pause_multiple_task_queues(
+        &self,
+        keys: Vec<TaskQueueKey>,
+        task_id: TaskIdentifier,
+        task: TaskDescriptor,
+    ) -> Result<ProposalWaiter, StateError> {
+        backfill_trace_field("queue_keys", format!("{keys:?}"));
```

Comment:
nit: you can set the `queue_keys` trace field in the `#[instrument]` macro like this:
```suggestion
    #[instrument(name = "propose_preempt_task_queues", skip_all, err, fields(queue_keys = ?keys, task_id = %task_id, task = %task.display_description()))]
    pub async fn pause_multiple_task_queues(
        &self,
        keys: Vec<TaskQueueKey>,
        task_id: TaskIdentifier,
        task: TaskDescriptor,
    ) -> Result<ProposalWaiter, StateError> {
```